enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,769 @@
+	// +kubebuilder:default:={openshift-*,kube-*,hypershift-*}
+	ExcludedNamespaces []string `json:"excludedNamespaces,omitempty"`
+    // +kubebuilder:default:=true
+	SkipTLSVerification bool `json:"skipTLSVerification,omitempty"`
+}
+
+// PodPlacementConfigStatus defines the observed state of PodPlacementConfig
+type PodPlacementConfigStatus struct {
+	// Conditions represents the latest available observations of a PodPlacementConfig's current state.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// PodPlacementConfig is the Schema for the podplacementconfigs API
+type PodPlacementConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PodPlacementConfigSpec   `json:"spec,omitempty"`
+	Status PodPlacementConfigStatus `json:"status,omitempty"`
+}
+
+```
+
+```yaml= 
+
+apiVersion: multiarch.openshift.io/v1alpha1
+kind: PodPlacementConfig
+metadata:
+  name: cluster
+spec:
+  logVerbosity: 1
+  skipTlsVerification: true
+  excludedNamespaces:
+    - openshift-*
+    - kube-*
+    - hypershift-*
+```
+
+### Implementation Details/Notes/Constraints
+As depicted in the following diagram, the multiarch-manager-operator consists of
+the following components:
+- `SchedulingGateMutatingWebhook`: it is a stateless component, registered as a
+  MutatingWebhook
+- `PodPlacementController`: it is the central controller of the operator. Given
+  a pod having the scheduling gate, it builds a set of pod-compatible
+  architectures and patches the pod's node affinity so that it is scheduled on
+  nodes having a CPU architecture supported by all the images the pod consists
+  of
+- `PodPlacementConfigController`: both the `SchedulingGateMutatingWebhook` and
+  the `PodPlacementController` consume the configuration information provided by
+  admins via the `PodPlacementConfig` CR to mutate the pod and set node affinity
+  through image inspection. The `PodPlacementConfigController` is run by the
+  operator to roll out the deployments of the business-logic components
+- `ImageFacade`: expose an interface hiding the location and nature of the image
+  metadata. The `getCompatibleArchitectureSet(image)` method will return the set
+  of compatible architectures for a given image. `ImageFacade` uses the
+  libraries in `github.com/containers/image`
+- `ImageFacadeProxyCache`: it implements an in-memory cache to reduce the
+  requests to `ImageFacade` and the external registries following a least
+  frequently used cache-eviction policy
+- `ImageRegistryController`: it watches the `image.config.openshift.io` and
+  adapts the information provided by its singleton object such that the
+  registries and image verification related cluster configuration can
+  be synced and used by the `RemoteRegistryInspector`
+- `RegistriesMirrorCacher`: it is a controller watching the
```

Comment:
Thus the controller would only see one of the supported architectures?

Reply:
not sure what you mean...this is intended to cover the case where ICSPs are used for mirroring.

Reply:
Got it